qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -572,12 +573,15 @@ def update_toolbar_waterfall_plot(self, is_waterfall):
         self.toolbar.set_generate_plot_script_enabled(not is_waterfall)
 
     def change_line_collection_colour(self, colour):
-        for col in self.canvas.figure.get_axes()[0].collections:
-            if isinstance(col, LineCollection):
+        axes = self.canvas.figure.get_axes()[0]
+        ax_type = axes_type(axes)
+        if ax_type == FigureType.Wireframe:
+            for col in axes.collections:
```

Comment:
This line does not check if the collection is a `LineCollection`, which maybe could mean that in the future the method `.set_color()` is called on a different object that does not support it?